File: zeppelin-examples/zeppelin-example-highcharts/src/main/resources/example/app/highcharts/highcharts.js
/*
 * Licensed to the Apache Software Foundation (ASF) under one or more
 * contributor license agreements.  See the NOTICE file distributed with
 * this work for additional information regarding copyright ownership.
 * The ASF licenses this file to You under the Apache License, Version 2.0
 * (the "License"); you may not use this file except in compliance with
 * the License.  You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

var data = [];
_.forEach($z.result.columnNames, function(col, series) {
  if (series == 0) return;
  var values = _.map($z.result.rows, function(row) {
    return {
      label: row[0],
      value : parseFloat(row[series])
    }
  });
  
  data.push({
    key : col.name,
    values : values
  })
});

jQuery.when(
  jQuery.getScript('https://code.highcharts.com/highcharts.js'),
  jQuery.getScript('https://code.highcharts.com/modules/exporting.js'),
  jQuery.getScript('https://cdn.rawgit.com/sdecima/javascript-detect-element-resize/master/jquery.resize.js'),
  jQuery('#highcharts_' + $z.id + ' svg').ready,
  jQuery.Deferred(function( deferred ){
    jQuery( deferred.resolve );
  })
).done(function() {
   jQuery('#highcharts_' + $z.id)
   .highcharts({
    title: {
      text: 'Monthly Average Temperature',
      x: -20 //center
    },
    subtitle: {
      text: 'Source: WorldClimate.com',
      x: -20
    },
    xAxis: {
      categories: ['Jan', 'Feb', 'Mar', 'Apr', 'May', 'Jun', 'Jul', 'Aug', 'Sep', 'Oct', 'Nov', 'Dec']
    },
    yAxis: {
      title: {
      text: 'Temperature (°C)'
    },
    plotLines: [{
      value: 0,
      width: 1,
      color: '#808080'
      }]
    },
    tooltip: {
      valueSuffix: '°C'
    },
    legend: {
      layout: 'vertical',
      align: 'right',
      verticalAlign: 'middle',
      borderWidth: 0
    },
    series: [{
      name: 'Tokyo',
      data: data
    }]
  });
});

File: zeppelin-examples/zeppelin-example-highcharts/src/main/java/org/apache/zeppelin/example/app/highcharts/HighCharts.java
/*
 * Licensed to the Apache Software Foundation (ASF) under one or more
 * contributor license agreements.  See the NOTICE file distributed with
 * this work for additional information regarding copyright ownership.
 * The ASF licenses this file to You under the Apache License, Version 2.0
 * (the "License"); you may not use this file except in compliance with
 * the License.  You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package org.apache.zeppelin.example.app.highcharts;

import org.apache.commons.io.IOUtils;
import org.apache.zeppelin.helium.Application;
import org.apache.zeppelin.helium.ApplicationContext;
import org.apache.zeppelin.helium.ApplicationException;
import org.apache.zeppelin.interpreter.InterpreterResult;
import org.apache.zeppelin.interpreter.dev.ZeppelinApplicationDevServer;
import org.apache.zeppelin.resource.*;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

import java.io.IOException;
import java.io.InputStream;
import java.text.SimpleDateFormat;
import java.util.Date;
import java.util.Random;

/**
 * Highcharts example: basic line graph
 * Input: the example inputs are hardcoded(?) in resources/examples/app/highcharts/highcharts.js
 */
public class HighCharts extends Application {
  private final Logger logger = LoggerFactory.getLogger(HighCharts.class);

  InterpreterResult result;

  public HighCharts(ApplicationContext context) {
    super(context);
  }

  @Override
  public void run(ResourceSet resources) throws ApplicationException, IOException {
    // Get data from resource args
    result = (InterpreterResult) resources.get(0).get();

    // create element
    println(String.format(
      "<div id=\"highcharts_%s\" style=\"height:400px\"><svg></svg></div>",
      context().getApplicationInstanceId()));
    // write js
    printResourceAsJavascript("example/app/highcharts/highcharts.js");
  }

  @Override
  public void unload() throws ApplicationException {
  }

  /**
   * Development mode
   */
  public static void main(String[] args) throws Exception {
    LocalResourcePool pool = new LocalResourcePool("dev");
    InputStream ins = ClassLoader.getSystemResourceAsStream(
      "example/app/highcharts/highcharts_mockdata.txt");
    InterpreterResult result = new InterpreterResult(
      InterpreterResult.Code.SUCCESS,
      InterpreterResult.Type.TABLE,
      IOUtils.toString(ins));
    pool.put(WellKnownResourceName.ZeppelinTableResult.name(), result);

    ZeppelinApplicationDevServer devServer = new ZeppelinApplicationDevServer(
      HighCharts.class.getName(),
      pool.getAll());

    devServer.start();
    devServer.join();
  }
}
